### PR TITLE
 Update FRE version for 2025 release

### PR DESCRIPTION
**Pull request by singhd789**
## Describe your changes
Just updating the FRE version 

## Issue ticket number and link (if applicable)
N/A 

## Checklist before requesting a review

- [ ] I ran my code
- [ ] I tried to make my code readable
- [ ] I tried to comment my code
- [ ] I wrote a new test, if applicable
- [ ] I wrote new instructions/documentation, if applicable
- [ ] I ran pytest and inspected it's output
- [ ] I ran pylint and attempted to implement some of it's feedback

### CODE REVIEW

File: docs/usage/yaml_framework.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-In order to utilize FRE 2024.01 tools, a distrubuted YAML structure is required. This framework includes a main model yaml, a compile yaml, a platforms yaml, and post-processing yamls. Throughout the compilation and post-processing steps, combined yamls that will be parsed for information are created. Yamls follow a dictionary-like structure with ``[key]: [value]`` fields. 
+In order to utilize FRE 2025.01 tools, a distrubuted YAML structure is required. This framework includes a main model yaml, a compile yaml, a platforms yaml, and post-processing yamls. Throughout the compilation and post-processing steps, combined yamls that will be parsed for information are created. Yamls follow a dictionary-like structure with ``[key]: [value]`` fields.
```
Comment (ceblanton) [inline docs/usage/yaml_framework.rst:1]: I bet we're going to miss updating some of these for 2025.02 and so on.

Maybe rephrase as "In order to utilize FRE tools"

File: docs/usage/yaml_framework.rst [hunk 2]
```diff
@@ -57,10 +57,10 @@ Where each dash indicates a list.
 
 .. code-block:: 
 
-  &version "2024.01"
+  &version "2025.01"
```
Comment (ceblanton) [inline docs/usage/yaml_framework.rst:60]: This is only an example of reusable variables, right?

Reusing the fre release number here, and having it be called "version" might give users that it's actually used by FRE do so something. Plus, we'll forget to update it.

File: fre/tests/test_fre_cli.py [hunk 3]
```diff
@@ -24,11 +24,11 @@ def test_cli_fre_option_dne():
 
 def test_fre_version():
     ''' module import flavor of below cli test '''
-    assert '2024.01' == fre.version
+    assert '2025.01' == fre.version
```
Comment (ceblanton) [inline fre/tests/test_fre_cli.py:27]: I guess this is ok.. we'll have to update this test every release.

**Review by ceblanton — APPROVED** [review]
Thanks! It looks great. I fear we'll forget some of these updates every release, but we can cross that as we go.

File: fre/tests/test_fre_cli.py (hunk 3 above)
Comment (singhd789) [reply fre/tests/test_fre_cli.py:27]: Yeah I wasn't sure about this one, definitely wanted to check

File: docs/usage/yaml_framework.rst (hunk 2 above)
Comment (singhd789) [reply docs/usage/yaml_framework.rst:60]: Do you mean just doing this instead `&version "(fre release number)"`? Could also do a variation of `&version "xxxx.xx"`?

Comment (ceblanton) [reply docs/usage/yaml_framework.rst:60]: why have "version" as a user-defined variable example though? Users might think that FRE is using that "version" variable somehow which it is not. Especially if in the example it's 2025.01 users will think it's actually used.

How about another example showing the user-defined variable, "Start_year" maybe.

File: fre/tests/test_fre_cli.py (hunk 3 above)
Comment (ilaflott) [reply fre/tests/test_fre_cli.py:27]: updating the version explicitly is the point. if the version is not updated everywhere, the pipeline SHOULD prevent us from publishing.

Comment (ilaflott) [reply fre/tests/test_fre_cli.py:27]: like- if you are changing either this `assert` statement, or `fre.version`, and you forgot to change it elsewhere, the pipeline should absolutely bonk a dev on the head.

File: docs/usage/yaml_framework.rst (hunk 2 above)
Comment (ilaflott) [reply docs/usage/yaml_framework.rst:60]: > Users might think that FRE is using that "version" variable somehow which it is not. Especially if in the example it's 2025.01 users will think it's actually used.

regarding this statement only... https://github.com/search?q=repo%3ANOAA-GFDL%2Ffre-cli+fre.version&type=code

agree that the module version probably shouldn't be used as an instructive lesson in `yaml` configuration, though.

Comment (singhd789) [reply docs/usage/yaml_framework.rst:60]: > why have "version" as a user-defined variable example though? Users might think that FRE is using that "version" variable somehow which it is not. Especially if in the example it's 2025.01 users will think it's actually used.
> 
> How about another example showing the user-defined variable, "Start_year" maybe.

OH. Yes ok I see, I 100% agree. I can change that tomorrow

**Review by ilaflott — APPROVED** [review]
